### PR TITLE
Feature/new module input + config context

### PR DESCRIPTION
**Pull request by jlews**
## What's new?

- Added input control and associated components to module-new folder.
- No need for email/number input components, base input component now discriminates on type attribute.
- Also added global Armstrong config context.
- Includes styling and stories

## Ticket number(s) in JIRA (if internal)

ARM-XX

[board](https://rocketmakers.atlassian.net/jira/software/projects/HM/boards/172)

## Checklist

- [x] are your changes in Storybook?
- [x] are any breaking changes documented in `docs/migrating_from_oldstrong.md`?
- [x] does _everything_ have jsdoc?
- [x] is everything exported from index.ts?
- [x] are all new hooks added to `src/hooks/hooks.stories.mdx` or given their own docs in Storybook?
- [x] are all new SCSS mixins added to `src/theme/mixins.stories.mdx`?
- [x] are all new SCSS variables added to `src/theme/variables.stories.mdx`?

### CODE REVIEW

File: module-new/src/components/spinner/spinner.component.tsx
Comment (tomRocketmakers) [inline module-new/src/components/spinner/spinner.component.tsx:1]: Does this component need a story?

Comment (jlews) [reply module-new/src/components/spinner/spinner.component.tsx:1]: Yeah it probably does to be fair, it didn't seem to have one in the old `module` folder so I left it for now

File: module-new/src/components/statusWrapper/statusWrapper.component.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+  return (
+    <>
+      {globals.inputStatusPosition === 'left' && (
+        <Status
+          className={className}
+          error={shouldShowErrorIcon}
+          pending={pending}
+          errorIcon={globals.validationErrorIcon}
+          data-position="left"
```
Comment (tomRocketmakers) [inline module-new/src/components/statusWrapper/statusWrapper.component.tsx:52]: Could this be done with a check inside the data-position?

Comment (jlews) [reply module-new/src/components/statusWrapper/statusWrapper.component.tsx:52]: Not sure I understand

Comment (tomRocketmakers) [reply module-new/src/components/statusWrapper/statusWrapper.component.tsx:52]: Never mind, was misreading how it was working

**Pull request merged**